src/main/java/org/spongepowered/api/event/item/inventory/EnchantItemEvent.java

```diff
@@ -0,0 +1,126 @@
+import org.spongepowered.api.event.item.inventory.enchantment.EnchantmentList;
+import org.spongepowered.api.event.item.inventory.enchantment.LevelRequirement;
+import org.spongepowered.api.item.inventory.ItemStackSnapshot;
+import org.spongepowered.api.item.inventory.Slot;
+
+/**
+ * This event is triggered when an item is enchanted at an enchanting table.
+ */
+public interface EnchantItemEvent extends InteractContainerEvent {
+
+    /**
+     * Returns the slot of the enchanting item.
+     *
+     * @return the slot of the enchanting item.
+     */
+    Slot getEnchantingSlot();
+
+    /**
+     * The seed for pseudo random enchantment generation.
+     *
+     * @return The seed
+     */
+    int getSeed();
+
+    /**
+     * The enchantment option. 1, 2 or 3
+     *
+     * @return The enchantment option
+     */
+    int getOption();
+
+    /**
+     * The itemstack to enchant.
+     *
+     * @return the itemstack to enchant.
+     */
+    ItemStackSnapshot getItem();
+
+    /**
+     * Triggers when the enchantment level requirement for an item is calculated.
+     */
+    interface CalculateLevelRequirement extends EnchantItemEvent {
+
+        /**
+         * The enchantment power based on bookshelves around the enchantment table.
+         * <p>In vanilla the maximum value is 15.</p>
+
+         * @return the enchantment power
+         */
+        int getPower();
+
+        /**
+         * The calculated level requirement for the option.
+         * <p>In vanilla the maximum value is 30.</p>
+         * <p>Returning a much higher level may result in no enchantments
+         * because the existing enchantments are too weak.</p>
+         *
+         * @return the level requirement.
+         */
+        Transaction<LevelRequirement> getLevelRequirement();
+
+    }
+
+    /**
+     * Triggers when the enchantments for an item are calculated.
+     * This happens after a new item to enchant is put in the table
+     * and again when the item is enchanted.
+     * <p>Note that when modifying the enchantment list you should
+     * return the same enchantments for the same seed and option.</p>
+     */
+    interface CalculateEnchantment extends EnchantItemEvent {
+
+        /**
+         * The final level requirement from {@link CalculateLevelRequirement}.
+         * <p>In vanilla the maximum value is 30.</p>
+         *
+         * @return the level requirement for the option
+         */
+        int getLevelRequirement();
+
+        /**
+         * The list of enchantments to apply to the item.
+         * <p>The first item in the list is used for display when previewing the enchantments.</p>
+         *
+         * @return the list of enchantments
+         */
+        Transaction<EnchantmentList> getEnchantments();
+
+    }
+
+    /**
+     * Triggers when the item was enchanted.
+     */
+    interface EnchantItem extends EnchantItemEvent, ChangeInventoryEvent {
```

Comment:
Maybe call this Post? The docs imply the action occurred, the name of the event implies it is about to occur.